pandas/core/indexes/period.py

```diff
@@ -210,6 +211,7 @@ def __new__(
         dtype: Dtype | None = None,
         copy: bool = False,
         name: Hashable = None,
+        is_period: bool = True,
```

Comment:
I don't think we want to add this as an argument - shouldn't this always be `True` here? If so, let's just assign `is_period=True` within the function body

Reply:
Thanks, I agree, `is_period` should be `True` for PeriodArray. I moved `is_period` into the function body.
Should we have `is_period` for the class PeriodIndex and assign it `True` as well?

Reply:
if `PeriodIndex` calls `to_offset` as well, then yes, that sounds reasonable